### DISCUSSION TITLE
Where is TurbSim called and its inputs defined?

### DISCUSSION THREAD

**Discussion by mlidGWT** [NLRWindSystems/WEIS#246] [Q&A]
Hello,

I'm trying to define a TurbSim grid that is a certain size, larger than the apparent default grid size, to avoid some out of bounds errors I'm getting in some of my simulations. I'm using the IEA 15 MW RWT and it seems that somewhere the grid is defined to be 300m x 300m, though I cannot for the life of me find where that is. I have tried using the following code in my `modeling_options.yaml` file, but it does not seem to have any effect:
```
turbulent_wind:    
   flag: True  
   HubHt: 150.  
   GridHeight: 350. 
   GridWidth: 350.
```

I don't see any grid dimension inputs in the OpenFAST `InflowWind.dat` file, which I am defining through a directory, though perhaps there is something in there that is precluding the above code from taking effect? It's hard to tell what WEIS will and won't overwrite with the OF directory method of running. (The input file has `WindType = 1` which _does_ get overwritten to `WindType = 3`, so that doesn't seem to be preventing this update from happening).

For reference, this is the error message I'm getting, even with the above block of code:
>FAST_Solution:FAST_AdvanceStates:SolveOption2c_Inp2AD_SrvD:InflowWind_CalcOutput:CalculateOutput:I
 fW_FlowField_GetVelAcc:Grid3DField_GetCell: GF wind array boundaries violated: Grid too small in
 Y direction. Y=-151.03; Y boundaries = [-150, 150]

Before this error, OpenFAST output this, so the timestamp confirms it was a new file written today and not accidentally grabbing an old file from before I tried to define the dimensions:
> Running InflowWind.

    Reading a 25x25 grid (300 m wide, 5.18799E-04 m to 300 m above ground) with a characteristic
    wind speed of 50 m/s. This full-field file was generated by TurbSim (from
    OpenFAST-v1.0-286-gcc303778) on 26-Oct-2023 at 13:21:55.

    Processed 5200 time steps of 20-Hz grid-field data (period of 260 seconds).

Any help finding where the current TurbSim wind field dimensions are being defined is greatly appreciated!

Best,
Meg

**Comment by dzalkind** [comment]
Hi Meg,

I have observed this behavior as well.  I have fixed it in a feature branch that still needs to be merged into WEIS.  We are only just starting a new phase of active development.

Generally, to find where things are set in WEIS, I search the code for, e.g., `GridWidth` to see where it's set and trace it through the code.  Our goal is to eventually eliminate the need for this kind of searching so that users don't have to be that familiar with the source code, but it will be an ongoing process over the next few years.

I hope this helps.  Thank you for all of your suggestions.

Best, Dan

**Comment by gbarter** [comment]
We'll circle up and get Dan's fixes incorporated.

To answer your question from an OpenFAST pespective, there should be a `wind` directory created where all of your OpenFAST files are.  Inside of that `wind` directory, if you look in the `.in` files, you will find all of the TurbSim inputs, including the domain size which you asked about.
